File: Projects/Project-Euler/P23.js
() => {
  let
  limit = 28123,
  abundants = [],
  sieve = new Array(limit).fill(true),
  sumsOfDivisors = new Array(limit).fill(1);
  result = 1;

  for (let i = 2; i <= limit; i++) {
    for (let j = i + i; j < limit; j += i) {
      sumsOfDivisors[j] += i;
    }
    if (sumsOfDivisors[i] > i && sumsOfDivisors[i] < limit) {
      abundants.push(i);
      abundants.map(abundant => {
        sieve[abundant + i] = false;
      });
    }
    result += sieve[i] && i;
  }
  return result;
}

File: Projects/Project-Euler/P23.py
target = 28123
abundants = []
sieve = list(range(0, target))
sum_divisors = [1] * target
result = 1

for i in range(2, target):
  for j in range(i + i, target, i):
    sum_divisors[j] += i

  if i < sum_divisors[i]:
    abundants.append(i)
    for abundant in abundants:
      if abundant + i < target:
        sieve[abundant + i] = 0
  
  result += sieve[i]

print(result)